httpx/_client.py

```diff
@@ -705,7 +667,8 @@ def request(
 
         ```python
         request = client.build_request(...)
-        response = client.send(request, ...)
+        with client.send(request, ...) as response:
+            response.read()
```

Comment:
I think _this_ might be worth discussing.

If we give folks `client.send()` and this returns a `response` context manager, shouldn't they expect `response` to be readily-usable, eg `response.text` returns the text right away?

Why would we want `.send()` to return a context manager, if we don't _actually_ need that interface except in the streaming case? Having to add `response.read()` each time as boilerplate seems like a smell, right?

Also, how do we help users understand what the effect of _not_ calling `response.read()` would be?

Given the pre-existing Requests model of ".request() gives me a response with the body readily available", I wonder about the implications of this.